tests/benchmark/src/main/java/org/jgroups/raft/DataReplicationBenchmark.java

```diff
@@ -0,0 +1,155 @@
+import org.jgroups.protocols.raft.RAFT;
+import org.jgroups.raft.testfwk.RaftTestUtils;
+import org.jgroups.util.Util;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+/**
+ * Benchmark the replication throughput.
+ *
+ * @author José Bolina
+ * @since 1.0.13
+ */
+@BenchmarkMode({Mode.Throughput})
+@Warmup(iterations = 10, time = 5)
+@Measurement(iterations = 5, time = 15)
+@Fork(value = 1, jvmArgsPrepend = "-Djava.net.preferIPv4Stack=true -Djgroups.udp.ip_ttl=0")
+public class DataReplicationBenchmark {
+
+    @Benchmark
+    public byte[] testReplication(ClusterState state) throws Exception {
+        return state.leader.set(state.data, 0, state.dataSize);
+    }
+
+    @State(Scope.Benchmark)
+    public static class ClusterState {
+
+        // Change the majority between values.
+        @Param({"1", "3", "5"})
+        public int clusterSize;
+
+        // Keep a factor of 4 between values.
+        @Param({"256", "1024", "4096"})
+        public int dataSize;
+
+        public byte[] data;
+
+        public RaftHandle[] members;
+
+        public RaftHandle leader;
+
+
+        @Setup
+        public void initialize() throws Exception {
+            List<String> memberList = IntStream.range(0, clusterSize)
+                    .mapToObj(i -> Character.toString('A' + i))
+                    .collect(Collectors.toList());
+
+            members = new RaftHandle[clusterSize];
+            for (int i = 0; i < clusterSize; i++) {
+                String name = Character.toString('A' + i);
+
+                // Utilize the default configuration shipped with jgroups-raft.
+                JChannel ch = new JChannel("raft.xml");
+                ch.name(name);
+
+                // A no-op state machine.
+                RaftHandle handler = new RaftHandle(ch, new EmptyStateMachine(dataSize));
+                RAFT raft = handler.raft();
+
+                // Default configuration.
+                raft.raftId(name);
+                raft.members(memberList);
+
+                // Fine-tune the RAFT protocol below.
+                raft.logClass(InMemoryLog.class.getCanonicalName());
+                raft.logUseFsync(false);
```

Comment:
This should use a field parameter?

Reply:
I updated it and left the `FileBasedLog` instead.